backend/main-service/src/main/java/kkakka/mainservice/coupon/util/PromotionScheduler.java

```diff
@@ -0,0 +1,77 @@
+package kkakka.mainservice.coupon.util;
+
+import java.time.LocalDateTime;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Optional;
+import kkakka.mainservice.coupon.application.CouponService;
+import kkakka.mainservice.coupon.application.DiscountService;
+import kkakka.mainservice.coupon.domain.Discount;
+import kkakka.mainservice.coupon.domain.repository.DiscountRepository;
+import kkakka.mainservice.coupon.ui.dto.CouponRequestDto;
+import kkakka.mainservice.member.member.domain.Grade;
+import kkakka.mainservice.member.member.domain.Member;
+import kkakka.mainservice.member.member.domain.repository.MemberRepository;
+import kkakka.mainservice.order.domain.repository.OrderRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class PromotionScheduler {
+
+    private final DiscountRepository discountRepository;
+    private final DiscountService discountService;
+    private final MemberRepository memberRepository;
+    private final OrderRepository orderRepository;
+    private final CouponService couponService;
+
+    @Scheduled(cron = "0 0 0 * * *")
+    public void checkDiscountExpiredDateSchedule() {
+        List<Discount> discounts = discountRepository.findAll();
+        for (Discount discount : discounts) {
+            if (LocalDateTime.now().isBefore(discount.getExpiredAt())) {
+                discountService.deleteDiscount(discount.getId());
+            }
+        }
+    }
+
+    @Scheduled(cron = "0 0 0 1 * *")
+    public void updateGradeSchedule() {
+        updateGradeByTotalPrice(findMembersByGrade(Grade.BRONZE), 10000, Grade.SILVER);
+        updateGradeByTotalPrice(findMembersByGrade(Grade.SILVER), 20000, Grade.GOLD);
+        updateGradeByTotalPrice(findMembersByGrade(Grade.GOLD), 30000, Grade.VIP);
+        updateGradeByTotalPrice(findMembersByGrade(Grade.VIP), 40000, Grade.VIP);
+        createGradeCoupon(Grade.BRONZE);
+        createGradeCoupon(Grade.SILVER);
+        createGradeCoupon(Grade.GOLD);
+        createGradeCoupon(Grade.VIP);
+    }
+
+    private List<Member> findMembersByGrade(Grade grade) {
+        return memberRepository.findByGrade(grade);
+    }
+
+    private void updateGradeByTotalPrice(List<Member> members, int price, Grade grade) {
+        for (Member member : members) {
+            Optional<Integer> totalPrice = orderRepository.findOrderPriceByMemberId(member.getId());
```

Comment:
```suggestion
            int totalPrice = orderRepository.findOrderPriceByMemberId(member.getId()).orElseGet(() -> 0);
```

`Optional` 처리가 필요해보여용